quinn-proto/src/connection/streams/mod.rs

```diff
@@ -180,8 +181,9 @@ impl<'a> RecvStream<'a> {
 
         // Clean up state after application observes the reset, since there's no reason for the
         // application to attempt to read or stop the stream once it knows it's reset
-        entry.remove_entry();
-        self.state.stream_freed(self.id, StreamHalf::Recv);
+        let (_, recv) = entry.remove_entry();
```

Comment:
I think that this path isn't covered by tests

Reply:
Improved coverage would be welcome, but doesn't need to be part of the PR, I think.